lib/saml_idp/service_provider.rb

```diff
@@ -20,13 +20,9 @@ def valid?
       attributes.present?
     end
 
-    def valid_signature?(doc, require_signature = false, options = {})
-      @matching_cert = Array(certs).find do |cert|
-        doc.valid_signature?(fingerprint_cert(cert), options.merge(cert:))
-      end
-
+    def valid_signature?(matching_cert = nil, require_signature = false)
       if require_signature || should_validate_signature?
-        !!@matching_cert
+        matching_cert.present?
```

Comment:
this is `matching_cert` the param name right? Should we remove `attr_reader :matching_cert` above to minimize confusion, esp now that `@matching_cert = ` has been removed?

Reply:
oh yeah my bad!

Reply:
fixed in 916e04346c3a2a42db2154ead80c20aea56f6e05